activestorage/app/models/active_storage/blob.rb

```diff
@@ -31,7 +31,7 @@ class ActiveStorage::Blob < ActiveStorage::Record
   # :method:
   #
   # Returns the associated ActiveStorage::Attachment instances.
-  has_many :attachments
+  has_many :attachments, inverse_of: false
```

Comment:
I do not understand why this change is necessary. It was motivated by [test suite failures](https://buildkite.com/rails/rails/builds/102569). 

Surprisingly, this change **does not** fix the test failures:

```diff
--- a/activestorage/app/models/active_storage/blob.rb
+++ b/activestorage/app/models/active_storage/blob.rb
@@ -31,7 +31,7 @@ class ActiveStorage::Blob < ActiveStorage::Record
   # :method:
   #
   # Returns the associated ActiveStorage::Attachment instances.
-  has_many :attachments, inverse_of: false
+  has_many :attachments, inverse_of: :blob
 
   ##
   # :singleton-method:
```

nor does defining `:inverse_of` for both sides of the association:

```diff
--- a/activestorage/app/models/active_storage/attachment.rb
+++ b/activestorage/app/models/active_storage/attachment.rb
@@ -30,7 +30,7 @@ class ActiveStorage::Attachment < ActiveStorage::Record
   # :method:
   #
   # Returns the associated ActiveStorage::Blob.
-  belongs_to :blob, class_name: "ActiveStorage::Blob", autosave: true
+  belongs_to :blob, class_name: "ActiveStorage::Blob", autosave: true, inverse_of: :attachments
 
   delegate_missing_to :blob
   delegate :signed_id, to: :blob
diff --git a/activestorage/app/models/active_storage/blob.rb b/activestorage/app/models/active_storage/blob.rb
index f180853979..ef9c6e09c6 100644
--- a/activestorage/app/models/active_storage/blob.rb
+++ b/activestorage/app/models/active_storage/blob.rb
@@ -31,7 +31,7 @@ class ActiveStorage::Blob < ActiveStorage::Record
   # :method:
   #
   # Returns the associated ActiveStorage::Attachment instances.
-  has_many :attachments, inverse_of: false
+  has_many :attachments, inverse_of: :blob
 
   ##
   # :singleton-method:
```

Reply:
👋 I created a PR in https://github.com/rails/rails/pull/50800 to fix this. Hopefully it will unblock this change !

Reply:
I merged #50800